### PR TITLE
For SBD and DRBD, use the disks that have relevant role assigned

### PR DESCRIPTION
**Pull request by jsuchome**
Based on https://github.com/crowbar/barclamp-pacemaker/pull/204

Requires https://github.com/crowbar/crowbar-core/pull/161

### CODE REVIEW

**Comment by jsuchome** [comment]
Hm, javascript magic does not seem to work any more...

**Comment by jsuchome** [comment]
Regarding that javascript: the code here used to work some months ago, before the repo changes.  @tboerger could you see what is wrong there now? Was there any significant change since then?

File: crowbar_framework/app/assets/javascripts/barclamps/pacemaker/application.js [hunk 1]
```diff
@@ -18,12 +18,18 @@
...
+  var disks;
+
   function StonithNodeAgents(el, options) {
     this.root = $(el);
     this.html = {
       table_row: '<tr data-id="{0}" data-alias="{1}"><td>{1}</td><td><input type="text" class="form-control" value="{2}"/></td></tr>'
     };
 
+    if (disks === undefined) {
+      disks = this.root.data('disks');
```
Comment (tboerger) [inline crowbar_framework/app/assets/javascripts/barclamps/pacemaker/application.js]: You should just write `this.disks = this.root.data('disks');` and drop the `var disks;` above.

File: crowbar_framework/app/assets/javascripts/barclamps/pacemaker/application.js [hunk 2]
```diff
@@ -141,7 +147,15 @@
 
     var rows = [];
     $.each(this.root.find('tbody tr'), function(index, tr) {
-      rows.push([$(tr).data('id'), $(tr).data('alias'), $(tr).find('input').val()]);
+      if (disks != undefined && disks[$(tr).data('id')] != undefined &&
```
Comment (tboerger) [inline crowbar_framework/app/assets/javascripts/barclamps/pacemaker/application.js]: `self.disks`

**Comment by tboerger** [comment]
> Regarding that javascript: the code here used to work some months ago, before the repo changes. @tboerger could you see what is wrong there now? Was there any significant change since then?

In general there are no changes.

**Comment by aspiers** [comment]
Please can you look at the issues CodeClimate found?  Also rebase to reduce the risk of gating failures due to [sync drift](https://bugzilla.suse.com/show_bug.cgi?id=935462).

**Comment by jsuchome** [comment]
There was no chance to develop this idea into usable state

**Pull request closed without merging**